controller/tap/server.go

```diff
@@ -254,15 +258,18 @@ func destinationLabels(resource *public.Resource) map[string]string {
 // of maxRps * 1s at most once per 1s window.  If this limit is reached in
 // less than 1s, we sleep until the end of the window before calling Observe
 // again.
-func (s *server) tapProxy(ctx context.Context, maxRps float32, match *proxy.ObserveRequest_Match, addr string, events chan *public.TapEvent) {
+func (s *server) tapProxy(maxRps float32, match *proxy.ObserveRequest_Match, addr string, events chan<- *public.TapEvent, wait *sync.WaitGroup) {
 	tapAddr := fmt.Sprintf("%s:%d", addr, s.tapPort)
 	log.Infof("Establishing tap on %s", tapAddr)
+	ctx := context.Background()
```

Comment:
Without propagating the context from the client request, how do the proxy tap requests ever get cancelled?

Reply:
From my reading of the [gRPC GoDocs](https://godoc.org/google.golang.org/grpc#DialContext) DialContext uses`ctx` the set up of the connection and the `ctx` struct doesn't manage the lifetime of the connection. IIUC, that means there is no need for the proxy tap requests to use the parent `ctx` provided by the `TapByResource`. The thought process is that each connection to a data plane tap should have its own context and when the connection is closed the context is canceled with it.

Reply:
I am not sure what the difference is between `Dial` and `DialContext` but we could probably get away with just using `Dial` when creating a connection since it provides its own context.